node/src/proxy_server/mod.rs

```diff
@@ -1228,6 +1235,44 @@ impl Hostname {
         };
         Self { hostname }
     }
+
+    fn is_valid(&self) -> Result<(), String> {
```

Comment:
The name `is_valid` isn't particularly well chosen. These names with "is" imply usually a boolean to be on the function's return.  

Maybe you want something like just "validate".